src/components/RegionSettings.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { MouseEvent } from "react";
+import { v4 as uuidv4 } from "uuid";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import Card from "@material-ui/core/Card";
+import CardActions from "@material-ui/core/CardActions";
+import CardContent from "@material-ui/core/CardContent";
+import RadioButtonCheckedIcon from "@material-ui/icons/RadioButtonChecked";
+import RadioButtonUncheckedIcon from "@material-ui/icons/RadioButtonUnchecked";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import ContinuousSlider from "./Slider";
+
+export default function RegionSettings() {
+  const [regions, setRegions] = React.useState<
+    Array<{
+      id: string;
+      type: string;
+      transparency: number;
+      activated: boolean;
+    }>
+  >([]);
+
+  const addNewRegion = (e: MouseEvent<HTMLButtonElement>) => {
+    setRegions([
+      ...regions,
+      { type: "None", transparency: 50, activated: false, id: uuidv4() },
+    ]);
+  };
+
+  const activateNewRegion = (id: string) => {
+    const regionToActivate = regions.filter(
+      (r) => r.id === id && r.type !== "None"
+    );
+    if (regionToActivate.length > 0) {
+      const newRegions = regions.map((r) =>
+        r.id === id ? { ...r, activated: true } : r
+      );
+      setRegions(newRegions);
+    }
+  };
+
+  const inactivateNewRegion = (id: string) => {
+    const newRegions = regions.map((r) =>
+      r.id === id ? { ...r, activated: false } : r
+    );
+    setRegions(newRegions);
+  };
+
+  const deleteRegion = (id: string) => {
+    const newRegions = regions.filter((r, i, rs) => r.id !== id);
+    setRegions(newRegions);
+  };
+
+  const setRegionType = (id: string, newType: string) => {
+    const newRegions = regions.map((r) =>
+      r.id === id ? { ...r, type: newType } : r
+    );
```

Comment:
Co myślisz o tym, żeby przerobić listę regionów na słownik `Record<RegionId, Region>`?

To by było czytelniejsze i wygodniejsze w użyciu ponieważ teraz za każdym razem, kiedy potrzebujesz zmienić konkretny region to musisz przejść przez całą listę. 

Natomiast gdyby był słownik, to by wygłądało trochę bardziej czytelnie:
```ts
setRegions({
   ...regions,
   [id]: {
      // New region description
    })